### PR TITLE
Remove unused code related to old http API  - Closes #1986

### LINKED ISSUE #1986: Remove unused code related to old http API
### What need to do 

We need to remove any unused code which is the traces of old HTTP API. 

```
helpers/http_api.js
helpers/router.js
```
Need to carefully see if any thing in use and remove unused code. 

### Which version(s) does this affect? (Environment, OS, etc...)
1.0.0

### PR DESCRIPTION
**Pull request by nazarhussain**
### What was the problem?
There were some unused code which appeared in code base by `0.9.x` versions. 

### How did I fix it?
Removed the unused code and tried to optimize the usage of existing code.

### How to test it?
Run the full test suit.

### Review checklist

* The PR solves #1986
* All new code is covered with unit tests
* All new code was formatted with Prettier
* Linting passes
* Tests pass
* Commit messages follow the [commit guidelines](CONTRIBUTING.md#git-commit-messages)
* Documentation has been added/updated

### CODE REVIEW

File: helpers/http_api.js [hunk 1]
```diff
@@ -14,10 +14,9 @@
...
 
-var _ = require('lodash');
-var extend = require('extend');
-var apiCodes = require('./api_codes');
+var queryParser = require('express-query-int');
 var checkIpInList = require('./check_ip_in_list');
+var apiCodes = require('./api_codes');
 
 /**
  * Description of the module.
```
Comment (diego-G) [inline helpers/http_api.js]: Please, update the doc header

File: test/unit/helpers/http_api.js [hunk 2]
```diff
@@ -397,404 +214,111 @@ describe('httpApi', () => {
...
 					},
-					api: sinonSandbox.stub(),
 				};
+				checkIpInListStub = sinonSandbox.stub();
+				httpApi.__set__('checkIpInList', checkIpInListStub);
 				done();
 			});
 
-			beforeEach(done => {
+			it('should response with error if api is not enabled', done => {
```
Comment (4miners) [inline test/unit/helpers/http_api.js]: `respond`

File: test/unit/helpers/http_api.js [hunk 3]
```diff
@@ -397,404 +214,111 @@ describe('httpApi', () => {
...
-					toJson: sinonSandbox.stub(),
-				};
-				done();
-			});
-
-			it('should call res.status(500).json() with error in json format', done => {
-				var tmp_error = validError.code
-					? validError.code
-					: apiCodes.INTERNAL_SERVER_ERROR;
+			it('should check for IP and response with success if api is not public and IP is whitelisted', done => {
```
Comment (4miners) [inline test/unit/helpers/http_api.js]: respond

File: test/unit/helpers/http_api.js [hunk 4]
```diff
@@ -397,404 +214,111 @@ describe('httpApi', () => {
...
 
-			describe('when response is empty', () => {
-				it('should call res.status with code = 204 and res.json', done => {
-					expect(validRes.status.calledWith(apiCodes.EMPTY_RESOURCES_OK)).to.be
-						.true;
-					expect(validRes.json.calledOnce).to.be.true;
-					done();
-				});
-			});
+			it('should check for IP and response with error if api is not public and IP is whitelisted', done => {
```
Comment (4miners) [inline test/unit/helpers/http_api.js]: same

**Pull request merged**